plugins/ingestion-kinesis/src/main/java/org/opensearch/plugin/kinesis/SequenceNumber.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.kinesis;
+
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.KeywordField;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermRangeQuery;
+import org.opensearch.index.IngestionShardPointer;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Kinesis sequence number.
+ */
+public class SequenceNumber implements IngestionShardPointer {
+
+    private final String sequenceNumber;
+    /** constant denoting non-existing sequence number */
+    public static final SequenceNumber NON_EXISTING_SEQUENCE_NUMBER = new SequenceNumber("non-existing-sequence-number");
+
+    /**
+     * Constructor
+     *
+     * @param sequenceNumber the sequence number
+     */
+    public SequenceNumber(String sequenceNumber) {
+        this.sequenceNumber = sequenceNumber;
+    }
+
+    /**
+     * Get the sequence number
+     *
+     * @return the sequence number
+     */
+    public String getSequenceNumber() {
+        return sequenceNumber;
+    }
+
+    @Override
+    public byte[] serialize() {
+        return sequenceNumber.getBytes(StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public String asString() {
+        return sequenceNumber;
+    }
+
+    @Override
+    public Field asPointField(String fieldName) {
+        return new KeywordField(fieldName, sequenceNumber, KeywordField.Store.YES);
+    }
```

Comment:
We discussed this on Slack. It sounds like rather than writing the sequence number as a string, we should encode it as a `byte[16]`, representing the underlying 128-bit number. I can take care of that in a follow-on PR.

Basically, the sequence number `String` is a 128-bit number printed out in decimal. Once all the digits are 9s, the next number will be `10^n` for some `n`, which will be lexicographically earlier, which would break the range queries. 

Instead, we can write the 128-bit number itself as big endian bytes, relying on the byte representation being lexicographically ordered as bytes.

Reply:
cool. thanks @msfroh for offering this help!